src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -161,6 +166,130 @@ message InfoResponse {
     optional string sdk_language_version = 4;
 }
 
+// === run_task_allocations_session RPC === //
+
+message SerializedObjectID {
+    optional string value = 1;
+}
+
+message SerializedObjectManifest {
+    optional SerializedObjectID id = 1;
+    optional SerializedObjectEncoding encoding = 2;
+    optional uint64 encoding_version = 3;
+    optional uint64 size = 4;
+    optional string sha256_hash = 5;
+}
+
+message SerializedObjectChunk {
+    // The ID of the serialized object this chunk belongs to.
+    optional SerializedObjectID id = 1;
+    // The chunk data.
+    optional bytes data = 2;
+}
+
+// Used for uploading a serialized object from initiating side to the accepting side.
+message UploadSerializedObjectRequest {
+    oneof message {
+        SerializedObjectManifest manifest = 1;
+        SerializedObjectChunk chunk = 2;
+    }
+}
+
+// The accepting side sends a response per request.
+message UploadSerializedObjectResponse {
```

Comment:
What is the point of responding to every chunk request? Grpc doesn't return a status for every message sent on a bi-directional stream?